### PR TITLE
Test Rail Listener Changes

### PR DESCRIPTION
**Pull request by kaushik691**
Modified Common Lib and created CustomTestRailListener also added respective docs

### CODE REVIEW

File: lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        """
+        response = self.api.suites.get_suites(project_id=project_id)
+        return response
+
+
+    def get_all_section_id(self, project_id, **kwargs):
+        """
+        Returns a list of sections for a project and test suite.
+
+        :param project_id:
```
Comment (souravDgraph) [inline lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py:110]: I see inconsistency some places \n is used. Some places it is not there.

File: lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
+"""
+ Test Rails lib implementation
+"""
+from typing import List
+
+from robot.api import logger
+from testrail_api import TestRailAPI
+
+# pylint: disable=C0301
```
Comment (souravDgraph) [inline lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py:9]: Why disabled??

File: lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        for section_details in sections:
+            if section_details['name'] == section_name:
+                section_id = section_details['id']
+        logger.debug(sections)
+        logger.info(section_id)
+        return section_id
+
+    def update_results(self, run_id, results: list):
+        """
+                This method expects an array of test results (via the 'results' field,
```
Comment (souravDgraph) [inline lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py:141]: Why there are tabs in documentations, and some places it is not there.

File: lib/test_rail_listener_lib/CustomTestRailListener/keywords/CustomTestRailListener/custom_test_rail_listener.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    exposed to the user through this library.
+    """
+    ROBOT_LIBRARY_SCOPE = 'TEST SUITE'
+    ROBOT_LIBRARY_VERSION = __version__
+    ROBOT_LISTENER_API_VERSION = 2
+
+    ROBOT_CODE_ROOT_FOLDER_PATH = None
+    ROBOT_CODE_BRANCH_FOLDER_PATH = None
+    ROBOT_CODE_UTILS_FOLDER_PATH = None
+    OS = None
```
Comment (souravDgraph) [inline lib/test_rail_listener_lib/CustomTestRailListener/keywords/CustomTestRailListener/custom_test_rail_listener.py:30]: where is this used??

**Review by souravDgraph — CHANGES_REQUESTED** [review]

File: lib/test_rail_listener_lib/CustomTestRailListener/keywords/test_rail_keywords.py [hunk 5]
```diff
@@ -0,0 +1,251 @@
+"""
+Test Rails Request Lib
+"""
+from CustomTestRailListener.components.test_rail_request import TestRailsRequest
+from robot.api import logger
+import os
+
+# pylint: disable=C0301
```
[the remark this answers is not in the document]
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/keywords/test_rail_keywords.py:8]: line too long

File: lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py (hunk 3 above)
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py:141]: few i copied it from the inbuilt documentation

File: test_suites/sample_test_cases/sample_test_rails.robot [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+Update Check for test rail
+    [Documentation]      Test case to add test run and update results
+    [Tags]  C1496   ${version}
+    ${proj_id}=     test rail get project id       Dgraph
+#    ${suite_id}=    test rail get suite id     ${proj_id}      Automation_Suite
+#    ${case_ids}      Create List     1496
+#    test rail add test run     ${proj_id}      suite_id=${suite_id}      name=Automation_Test_Run       description="Robot famrework check"     case_ids=${case_ids}
+#    ${run_id}=      test rail get run id by name   ${proj_id}      Automation_Test_Run     created_after=1609499627
+#    test rail add result for test case test run    ${run_id}   1496    status_id=2     version=20.11   comment=Updating Comments as part of Autamation comment
```
[the remark this answers is not in the document]
Comment (kaushik691) [reply test_suites/sample_test_cases/sample_test_rails.robot:18]: will make the change for username and password

File: lib/test_rail_listener_lib/setup.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+#####!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (kaushik691) [reply lib/test_rail_listener_lib/setup.py:1]: # <-okay

File: lib/test_rail_listener_lib/CustomTestRailListener/keywords/test_rail_keywords.py [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+        outfile.write("run_name: %s\n" % run_name)
+        outfile.write("run_id: %s\n" % self.run_id)
+        outfile.write("project_name: %s\n" % project_name)
+        outfile.write("project_id: %s\n" % self.project_id)
+        logger.debug("Initialized Test Rails for " + url)
+
+    def test_rail_get_all_tests(self, run_id):
+        """
+        Method to get all the tests created as part of run_id
+        \n:param run_id: 7
```
[the remark this answers is not in the document]
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/keywords/test_rail_keywords.py:56]: okay

File: lib/test_rail_listener_lib/CustomTestRailListener/keywords/CustomTestRailListener/custom_test_rail_listener.py (hunk 4 above)
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/keywords/CustomTestRailListener/custom_test_rail_listener.py:30]: will change it

File: lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py (hunk 2 above)
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py:9]: line too long

File: lib/test_rail_listener_lib/CustomTestRailListener/keywords/CustomTestRailListener/custom_test_rail_listener.py (hunk 4 above)
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/keywords/CustomTestRailListener/custom_test_rail_listener.py:30]: done

File: test_suites/sample_test_cases/sample_test_rails.robot (hunk 6 above)
[the remark this answers is not in the document]
Comment (kaushik691) [reply test_suites/sample_test_cases/sample_test_rails.robot:18]: done

File: lib/test_rail_listener_lib/setup.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (kaushik691) [reply lib/test_rail_listener_lib/setup.py:1]: done

File: lib/test_rail_listener_lib/CustomTestRailListener/keywords/test_rail_keywords.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/keywords/test_rail_keywords.py:56]: done

File: lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py (hunk 1 above)
Comment (kaushik691) [reply lib/test_rail_listener_lib/CustomTestRailListener/components/test_rail_request.py:110]: checked

**Pull request closed without merging**